src/d/d_mesg.cpp

```diff
@@ -4,460 +4,2244 @@
+    strcat(i_Msg->text[1], local_38);
+    strcat(i_Msg->text[2], local_2c);
+    strcat(i_Msg->text[3], local_38);
 }
 
-/* 801E6764-801E6768       .text setTextPosition__18dMesg_screenData_cFUc */
-void dMesg_screenData_c::setTextPosition(u8) {
-    /* Nonmatching */
-}
-
-/* 801E6768-801E676C       .text createScreen__18dMesg_screenData_cFv */
-void dMesg_screenData_c::createScreen() {
-    /* Nonmatching */
-}
-
-/* 801E676C-801E67C8       .text __dt__18dMesg_screenData_cFv */
-dMesg_screenData_c::~dMesg_screenData_c() {
-    /* Nonmatching */
+/* 801E62F8-801E6760       .text dMesg_waitProc__FP14sub_mesg_class */
+void dMesg_waitProc(sub_mesg_class* i_Msg) {
+    /* Nonmatching */
+    nowMesgCode = (dMesg_gpControl->mGroupID << 16) | dMesg_gpControl->mMessageIndex;
+    zenkaku = 0;
+    zenkakuCode = 0;
+    if (oldMesgCode != nowMesgCode) {
+        if (dMesg_hyrule_language_check((dMesg_gpControl->mGroupID << 16) | dMesg_gpControl->mMessageIndex)) {
+            nowMesgCode = (dMesg_gpControl->mGroupID << 16) | dMesg_gpControl->mMessageIndex;
+            if (!headerFlag) {
+                dMesg_gpControl->mMainFont = dMesg_gpRFont;
+                headerFlag = true;
+            }
+        } else {
+            if (headerFlag) {
+                dMesg_gpControl->mMainFont = dMesg_gpFont;
+                headerFlag = false;
+            }
+        }
+        i_Msg->field_0x100 = JKRCreateExpHeap(0xa32d, i_Msg->heap, false);
+        JKRHeap* heap = mDoExt_setCurrentHeap(i_Msg->field_0x100);
+        for (int i = 0; i < 4; i++) {
+            if (!i_Msg->text[i]) {
+                i_Msg->text[i] = (char*)i_Msg->field_0x100->alloc(1001, 4);
+                JUT_ASSERT(3304, i_Msg->text[i] != 0);
+            }
+        }
+        if (!i_Msg->screen) {
+            u32 r28 = nowMesgCode;
+            JMessage::TResource* resource = dMesg_gpControl->getResource_groupID(r28 >> 16);
+            void* messageData = resource == NULL ? NULL : resource->getMessageData_messageIndex(r28);
+            unk_messageData stack_3c = *(unk_messageData*)messageData;
+            if (stack_3c.field_0xc == 9) {
+                i_Msg->screen = new dMesg_screenDataItem_c();
+            } else {
+                i_Msg->screen = new dMesg_screenDataTalk_c();
+            }
+            JUT_ASSERT(3317, i_Msg->screen != 0);
+            i_Msg->screen->mMesg = i_Msg;
+            i_Msg->screen->mHeap = i_Msg->field_0x100;
+            if (headerFlag) {
+                i_Msg->screen->setFont(dMesg_gpRFont, dMesg_gpRFont);
+            } else {
+                i_Msg->screen->setFont(dMesg_gpFont, dMesg_gpRFont);
+            }
+            i_Msg->screen->createScreen();
+            i_Msg->screen->setTextPosition(stack_3c.field_0xe);
+            dMesg_gbUpdate = true;
+            i_Msg->field_0x164 = 2;
+            dMesg_gpSequenceProcessor->field_0x6c = -1;
+            mDoAud_talkIn();
+            mDoAud_seStart(0x80c);
```

Comment:
There's an enum with sound effect names, check JAZelAudio_SE.h. Also applies to other mDoAud_seStart calls in this file

Reply:
Fixed.